BondSource.cpp

```diff
@@ -0,0 +1,188 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2014 - 2020 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+#include <ibamr/config.h>
+
+#include <ibamr/app_namespaces.h>
+
+#include <HierarchyDataOpsManager.h>
+#include <SAMRAI_config.h>
+#include <math.h>
+// local include
+#include "BondSource.h"
+
+/////////////////////////////// PUBLIC ///////////////////////////////////////
+
+BondSource::BondSource(SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM>> phi_u_var,
+                    SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM>> phi_a_var,
+                    SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM>> w_var,
```

Comment:
We need some way to set `z_var` and `d_sigma_var`. I would suggest setting them in the constructor (in place of `w_var` which is no longer needed).